addons/purchase/models/purchase.py

```diff
@@ -1065,6 +1067,14 @@ def onchange_product_id(self):
         # Reset date, price and quantity since _onchange_quantity will provide default values
         self.price_unit = self.product_qty = 0.0
 
+        # remove packaging if not match the product
+        if self.product_packaging_id.product_id != self.product_id:
+            self.product_packaging_id = False
+        # add single unit packaging if no packaging
+        if not self.product_packaging_id:
+            single_unit_packagings = self.product_id.packaging_ids.filtered(lambda p: p.qty == 1)
```

Comment:
We speak about it with thd and we could find the more suitable packaging in any case. As example, if we have
Pack of 1
Pack of 6
Pallet of 30
 Then if the so line has
- 36 units we could suggest 6 pack of 6
- 37 units -> 37 pack of 1
- 30 units -> Pallet
The upper common divider